### PR TITLE
CAT-1302 Vertically centered cursor in FE

### PR DESCRIPTION
**Pull request by Fasusu**
Single TestClass Test is green:
https://jenkins.catrob.at/job/Catroid-CustomEmulator-Single-ClassAndPackage-Test/447/

### CODE REVIEW

File: catroid/src/org/catrobat/catroid/formulaeditor/FormulaEditorEditText.java [hunk 1]
```diff
@@ -193,12 +193,17 @@ protected void onDraw(Canvas canvas) {
 		paint.setStrokeWidth(3);
 
 		Layout layout = getLayout();
+
 		if (layout != null) {
+
 			int line = layout.getLineForOffset(absoluteCursorPosition);
+
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/formulaeditor/FormulaEditorEditText.java]: 3 lines of code, 4 empty lines... Remove *your* newly added empty lines.
Also https://wiki.openstack.org/wiki/GitCommitMessages#Things_to_avoid_when_creating_commits

File: catroid/src/org/catrobat/catroid/formulaeditor/FormulaEditorEditText.java [hunk 2]
```diff
@@ -193,12 +193,17 @@ protected void onDraw(Canvas canvas) {
 		paint.setStrokeWidth(3);
 
 		Layout layout = getLayout();
+
 		if (layout != null) {
+
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/formulaeditor/FormulaEditorEditText.java]: Let's see if there is a checkstyle rule for that :wink:

File: catroid/src/org/catrobat/catroid/formulaeditor/FormulaEditorEditText.java [hunk 3]
```diff
@@ -193,12 +193,17 @@ protected void onDraw(Canvas canvas) {
...
 			int line = layout.getLineForOffset(absoluteCursorPosition);
+
 			float xCoordinate = layout.getPrimaryHorizontal(absoluteCursorPosition) + getPaddingLeft();
 			float startYCoordinate = layout.getLineBaseline(line) + layout.getLineAscent(line);
 			float endYCoordinate = layout.getLineBaseline(line) + layout.getLineAscent(line) + getTextSize();
-			endYCoordinate += line == 0 ? 5 : 0; // First line in FE is a little bit higher so we need a bigger cursor too.
+
+			startYCoordinate += (line == 0) ? 12.5f : 5.0f; // add top offset for cursor to make it vertically centered
+			endYCoordinate += line == 0 ? 12.5f : 2.5f; // First line in FE is a little bit higher so we need a bigger cursor too.
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/formulaeditor/FormulaEditorEditText.java]: Inconsistent use of braces.
ping @cheidenreich: How will this be handled by the pmd controversial ruleset?

**Comment by Fasusu** [comment]
Will fix issues and discuss with UX and resubmit

**Comment by robertpainsi** [comment]
@Fasusu Any news?

**Pull request closed without merging**